src/PhpSessionPersistence.php

```diff
@@ -231,18 +228,15 @@ private function generateCacheHeaders() : array
     }
 
     /**
-     * Return the Last-Modified header line based on the request's script file
-     * modified time. If no script file could be derived from the request we use
-     * this class file modification time as fallback.
+     * Return the Last-Modified header line based on main script of execution
+     * modified time. If unable to get a valid timestamp we use this class file
+     * modification time as fallback.
      * @return string|false
      */
     private function getLastModified()
     {
-        if ($this->scriptFile && is_file($this->scriptFile)) {
-            return gmdate(self::HTTP_DATE_FORMAT, filemtime($this->scriptFile));
-        }
-
-        return false;
+        $lastmod = getlastmod() ?: filemtime(__FILE__);
```

Comment:
That is such a nice find - I'd never known about that, and it was the exact functionality I was trying to create previously!